staging/src/k8s.io/client-go/kubernetes/typed/core/v1/node_expansion.go

```diff
@@ -18,16 +18,72 @@ package v1
 
 import (
 	"context"
+	"fmt"
+	"strconv"
+	"strings"
 
 	v1 "k8s.io/api/core/v1"
+	"k8s.io/api/node/v1alpha1"
 	"k8s.io/apimachinery/pkg/types"
+	restclient "k8s.io/client-go/rest"
 )
 
 // The NodeExpansion interface allows manually adding extra methods to the NodeInterface.
 type NodeExpansion interface {
 	// PatchStatus modifies the status of an existing node. It returns the copy
 	// of the node that the server returns, or an error.
 	PatchStatus(ctx context.Context, nodeName string, data []byte) (*v1.Node, error)
+	GetLogs(nodeName string, opts *v1alpha1.NodeLogViewerOptions) *restclient.Request
+}
+
+// GetLogs constructs a request for getting the logs for a node
+func (c *nodes) GetLogs(nodeName string, opts *v1alpha1.NodeLogViewerOptions) *restclient.Request {
```

Comment:
I expected this to look much more like the pod logs implementation, encoding the versioned parameters, rather than piece-mill serialization into query parameters:

```
// Get constructs a request for getting the logs for a pod
func (c *pods) GetLogs(name string, opts *v1.PodLogOptions) *restclient.Request {
	return c.client.Get().Namespace(c.ns).Name(name).Resource("pods").SubResource("log").VersionedParams(opts, scheme.ParameterCodec)
}
```